composer/datasets/in_context_learning_evaluation.py

```diff
@@ -12,11 +12,10 @@
 import torch
 import transformers
 from torch.utils.data import DataLoader, Dataset
-from tqdm import tqdm
 
 from composer.core import DataSpec
 from composer.core.data_spec import _default_split_batch, _split_list
-from composer.utils import MissingConditionalImportError, dist, get_file
+from composer.utils import MissingConditionalImportError, dist, get_file, parse_uri
 
 if TYPE_CHECKING:
     import transformers
```

Comment:
Why do we reimport transformers here? It's already imported on line 13?

Reply:
optional dependencies are only supposed to be imported conditionally, as it is done here. however, it seems like that didn't happen for this file or got undone at some point (the import on line 13 is bad). If we had properly put these classes in the __init__.py, this would've made composer uninstallable without installing transformers. would be great if you could fix both of these things :)

Reply:
Should we conditionally import transformers in the test too, [here](https://github.com/mosaicml/composer/blob/5a15d68d589700d5afe3dd6f7e8ce0f17590d459/tests/datasets/test_in_context_learning_datasets.py#L12)?

Reply:
yup, should be imported with `pytest.importorskip('transformers')`

Reply:
Fixed in test file